configure.ac

```diff
@@ -452,9 +452,43 @@ if test "$enable_werror" = "yes"; then
   fi
   ERROR_CXXFLAGS=$CXXFLAG_WERROR
 
-  dnl -Wstringop-overread and -Wstringop-overflow are broken in gcc
-  AX_CHECK_COMPILE_FLAG([-Wstringop-overread], [WARN_CXXFLAGS="$WARN_CXXFLAGS -Wno-stringop-overread"], [], [$CXXFLAG_WERROR])
-  AX_CHECK_COMPILE_FLAG([-Wstringop-overflow], [WARN_CXXFLAGS="$WARN_CXXFLAGS -Wno-stringop-overflow"], [], [$CXXFLAG_WERROR])
+  dnl -Warray-bounds and -Wdangling-reference cause problems with GCC. Do not treat these warnings as errors.
+  AX_CHECK_COMPILE_FLAG([-Warray-bounds], [NOWARN_CXXFLAGS="$NOWARN_CXXFLAGS -Wno-error=array-bounds"], [], [$CXXFLAG_WERROR], [AC_LANG_SOURCE([
```

Comment:
Consider completely ignoring -Warray-bounds

```
commit 057e45fa39d6a7f8c03f627641be3b4998d89f14
Author: Konstantin Akimov <knstqq@gmail.com>
Date:   Sat Apr 26 00:51:31 2025 +0700

    fix: -Warray-bounds to ignore with gcc as it is incompatible with immer implementation
```

I have hundreds of warning and output is too noisy, difficult to spot any real error or warning